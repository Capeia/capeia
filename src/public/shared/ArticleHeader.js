// @flow
import React, { Component } from 'react'
import withStyles from 'isomorphic-style-loader/lib/withStyles'
import s from './ArticleHeader.scss'

class ArticleHeader extends Component {
  props: {
    thumbnail: $FlowFixMe,
    children?: $FlowFixMe
  };

  render () {
    const { thumbnail, children } = this.props
    return (
      <header className={s.articleHeader}>
        <div className={s.thumbnail}>
          {thumbnail}
        </div>
        <div className={s.heading}>
          {children}
        </div>
      </header>
    )
  }
}

export default withStyles(s)(ArticleHeader)
