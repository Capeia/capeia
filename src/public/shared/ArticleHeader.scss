@import '~theme/variables';

.articleHeader {
  display: flex;
  align-items: center;
  // prevent long titles from squishing the thumbnail
  .heading {
    flex: 1;
  }

  @media (max-width: $screen-xs-max) {
    justify-content: center;
    flex-wrap: wrap;

    .heading {
      flex: auto;
      margin-left: 0;
    }
  }
}

.thumbnail {
  @media print {
    // TODO: Get this to show properly in print
    display: none;
  }
}

.heading {
  margin-left: 1em;

  h1, :global(.h1) {
    margin: 0;
  }

  address {
    font-size: 1.2em;
  }
}
