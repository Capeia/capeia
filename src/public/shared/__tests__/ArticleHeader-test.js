// @flow
/* eslint-env jest, jasmine */

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ArticleHeader from '../ArticleHeader'

describe('<ArticleHeader>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ArticleHeader thumbnail={<span>Foo</span>}>Bar</ArticleHeader>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render a given thumbnail and children with correct wrappers', () => {
    const thumb = <span className='MyThumb' />
    const wrapper = shallow(<ArticleHeader thumbnail={thumb}>FooBar</ArticleHeader>)
    expect(wrapper.equals(
      <header className='articleHeader'>
        <div className='thumbnail'>
          {thumb}
        </div>
        <div className='heading'>
          FooBar
        </div>
      </header>
    )).toEqual(true)
  })
})
