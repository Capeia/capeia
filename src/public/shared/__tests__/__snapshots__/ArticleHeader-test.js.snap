// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeader> renders correctly 1`] = `
<header
  className="articleHeader"
>
  <div
    className="thumbnail"
  >
    <span>
      Foo
    </span>
  </div>
  <div
    className="heading"
  >
    Bar
  </div>
</header>
`;
